app/utils/routeResponseValidator.js

```diff
@@ -35,15 +35,18 @@ const buildValidatedJsonHandler = (controller, apiDoc) => {
     responses,
     components: commonApiDoc.components,
   })
+
   const handler = async function (req, res) {
-    const { status, response } = await controller(req)
+    const { status, response, headers } = await controller(req)
```

Comment:
I don't believe any code is using these enhancements now. Is there a reason to keep the additional complexity introduced here?

Reply:
yes, for handling dynamic headers, default is application/json so it does not break anything, as third argument is optional